snapcraft/internal/remote_build/_launchpad.py

```diff
@@ -44,28 +45,72 @@
 class LaunchpadClient:
     """Launchpad remote builder operations."""
 
-    def __init__(self, *, project: Project, build_id: str, user: str) -> None:
+    def __init__(
+        self,
+        *,
+        project: Project,
+        build_id: str,
+        user: str,
+        architectures: Optional[List[str]] = None,
+        git_branch: str = "master",
+        core18_channel: str = "stable",
+        snapcraft_channel: str = "edge",
+    ) -> None:
         if not Git.check_command_installed():
             raise errors.GitNotFoundProviderError(provider="Launchpad")
 
-        self._id = build_id
-        self._core18_channel = "stable"
-        self._snapcraft_channel = "edge"
-        self._name = project.info.name
-        self._waiting = []  # type: List[str]
-        self._data_dir = os.path.join(
-            BaseDirectory.save_data_path("snapcraft"), "launchpad"
-        )
-        self._cache_dir = os.path.join(
-            BaseDirectory.save_cache_path("snapcraft"), "launchpad"
-        )
+        self._snap_name = project.info.name
+        self._build_id = build_id
+
+        self.architectures = architectures
+        self.user = user
+
+        self._lp_name = build_id
+        self._lp_git_branch = git_branch
 
-        os.makedirs(self._data_dir, mode=0o700, exist_ok=True)
+        self._core18_channel = core18_channel
+        self._snapcraft_channel = snapcraft_channel
+
+        self._cache_dir = self._create_cache_directory()
+        self._data_dir = self._create_data_directory()
         self._credentials = os.path.join(self._data_dir, "credentials")
 
-        self._user = user
+        self._lp: Launchpad = None
+        self._waiting = []  # type: List[str]
+
+    @property
+    def architectures(self) -> List[str]:
+        return self._architectures
+
+    @architectures.setter
+    def architectures(self, architectures: List[str]) -> None:
```

Comment:
use Sequence instead of List https://docs.python.org/3.6/library/typing.html#typing.List

Reply:
Good point, will do.